subprojects/gradle/module-types/src/test/kotlin/com/avito/android/module_type/internal/ModulesRestrictionsFinderTest.kt

```diff
@@ -0,0 +1,127 @@
+package com.avito.android.module_type.internal
+
+import BetweenModuleTypes
+import DependentModule
+import FeatureModule
+import LibraryModule
+import TestModule
+import ToTestModule
+import com.avito.android.module_type.DependencyRestriction
+import com.avito.android.module_type.ModuleWithType
+import com.avito.module.configurations.ConfigurationType.Main
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+internal class ModulesRestrictionsFinderTest {
```

Comment:
Unit tests for logic.
They are similar to gradle tests but check only logic with prepared data.